TrackingTools/GsfTools/src/BasicMultiTrajectoryState.cc

```diff
@@ -5,133 +5,128 @@
 
 using namespace SurfaceSideDefinition;
 
-BasicMultiTrajectoryState::BasicMultiTrajectoryState( const std::vector<TSOS>& tsvec) :
-  BasicTrajectoryState(tsvec.front().surface()), theStates(tsvec) {
- 
+BasicMultiTrajectoryState::BasicMultiTrajectoryState(const std::vector<TSOS>& tsvec)
+    : BasicTrajectoryState(tsvec.front().surface()), theStates(tsvec) {
   // only accept planes!!
   const BoundPlane* bp = dynamic_cast<const BoundPlane*>(&tsvec.begin()->surface());
-  if UNLIKELY( bp==nullptr )
-	       throw cms::Exception("LogicError") << "MultiTrajectoryState constructed on cylinder";
-   
-  for (auto i=tsvec.begin(); i!=tsvec.end(); i++) {
-    if UNLIKELY(!i->isValid()) {
-      throw cms::Exception("LogicError") << "MultiTrajectoryState constructed with invalid state";
-    }
-    if UNLIKELY(i->hasError() != tsvec.front().hasError()) {
-      throw cms::Exception("LogicError") << "MultiTrajectoryState mixes states with and without errors";
-    }
-    if UNLIKELY( &i->surface() != &tsvec.front().surface()) {
-      throw cms::Exception("LogicError") << "MultiTrajectoryState mixes states with different surfaces";
-    }
-    if UNLIKELY( i->surfaceSide() != tsvec.front().surfaceSide()) {
-      throw cms::Exception("LogicError") 
-	<< "MultiTrajectoryState mixes states defined before and after material";
-    }
-    if UNLIKELY( i->localParameters().pzSign()*tsvec.front().localParameters().pzSign()<0. ) {
-      throw cms::Exception("LogicError") 
-	<< "MultiTrajectoryState mixes states with different signs of local p_z";
-    }
+  if
```

Comment:
The issue with the UNLIKELY macro has been pointed out already elsewhere, and I hope it will get fixed centrally at some time